### DISCUSSION TITLE
[2주차] HTTP 메소드의 멱등성, 그리고 안전한 메서드

### DISCUSSION THREAD

**Discussion by yunjjeongjo** [prgrms-web-devcourse/We-are-the-MASTERS-OF-HTTP#32] [Interview Question]
멱등성의 의미, 그리고 HTTP Method 에서의 멱등성과 안전한 메소드(Safe method)에 대해 설명해주세요!

**Comment by punchdrunkard** [comment]
멱등성이란 간단하게 말하자면, 몇 번을 실행해도 결과가 같음 을 말합니다.
HTTP 에서 멱등성을 따질 때는, **DB의 상태**를 기준으로 멱등성을 따집니다.
즉, HTTP 메서드가 멱등성을 가진다는 것은 **해당 요청에 대한 서버의 상태가 항상 같다**는 의미입니다.

안전한 메소드란 **서버의 상태를 변경시키지 않는 메소드**를 의미합니다.
`GET`이나 `HEAD`가 해당 메서드에 속합니다.

멱등성을 같지만 안전한 메서드가 아닌 `DELETE`를 예로 들자면,
`DELETE` 요청을 보내면 **삭제된 자원에 대한 response**는 항상 동일하기 때문에 `DELETE`는 멱등하지만,
서버에서 어떤 자원을 삭제하는 (서버의 상태를 변경시키는) 작업을 하기 때문에 안전한 메서드가 아닙니다.

ps 오타 고쳐드렸습니다 ㅎㅎ
